clappr/clappr-android
# Remove kotlin reflect

This is a big one
## Summary 

Currently, the `Loader` component uses reflection in order to manage plugin types. Since `kotlin-reflect` imposes a large size penalty to the player (compared to Clappr size), removing it should improve the bundle size and reduce some risks associated with using reflection.

## Changes
- Change Plugin/UIPlugin inheritance using interfaces
- Change Loader into a singleton
- Change Loader registration
- Removed MediaControl expecting to load plugins in the order they were registered.

## How to test

- Run tests on `clappr` module
- Run `app`
- Both should work as before, there shouldn't be any major issues.
